### PR TITLE
Blog App : Device

### PR DESCRIPTION
**Pull request by jeanbuhendwa**
- [ ] Removed current_user method in ApplicationController, devise will provide us with one.
- [ ] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- [ ] Can register a new user.
- [ ] User logs in with a combination of email and password.
- [ ] Hashed password should be stored in the database.
- [ ] Ask for confirmation of email.
- [ ] Can reset password.

### CODE REVIEW

File: db/migrate/20230620225226_add_devise_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (omar-muhamad) [inline db/migrate/20230620225226_add_devise_to_users.rb:28]: - [ ] Please uncomment these lines to make the `confirmable` method work as expected.
```suggestion
      ## Confirmable
       t.string   :confirmation_token
       t.datetime :confirmed_at
       t.datetime :confirmation_sent_at
      # t.string   :unconfirmed_email # Only if using reconfirmable
```

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -12,6 +12,8 @@
 
   <body>
     <main>
+      <p class="notice"><%= notice %></p>
+      <p class="alert"><%= alert %></p>
       <%= yield %>
     </main>
```
Comment (omar-muhamad) [inline app/views/layouts/application.html.erb:18]: - [ ] After I signed in this error was raised when the app tried to open the home page `user#index`, Please fix that.

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi Team,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by DeliceLydia — APPROVED** [review]
## Hi @jeanbuhendwa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**